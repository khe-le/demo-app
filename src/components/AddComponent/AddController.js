import React, { useState } from 'react'
import AddForm from './AddForm'
import AddIcons from './AddIcons'
import IslandStore from '../actions/store'

function AddController(){
    const [villagerImage, setVillagerImage] = useState('4.gif')
    
    function setRandomVillager(){
        const rand = Math.ceil(Math.random() * 10)
        setVillagerImage(`${rand}.gif`)
    }

    function addVillager(newVillager){
        newVillager.image = villagerImage
        IslandStore.addVillager(newVillager)
    }

    return (
        <React.Fragment>
                <AddIcons 
                    image = {villagerImage}
                    randomize={setRandomVillager}/>

                <AddForm 
                    image={villagerImage}
                    addVillager={addVillager}
                />
        </React.Fragment>

    )
}

export default AddController