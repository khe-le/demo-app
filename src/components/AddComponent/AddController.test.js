import React from 'react'
import AddController from './AddController'
import Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import faker from 'faker'

Enzyme.configure({ adapter: new Adapter() })

describe('<AddController />', () => {
    let wrapper = Enzyme.mount(<AddController/>)
    let randNumber = faker.random.number({ min: 0, max: 1}) 
    const originalConsoleError = console.error;

    beforeEach(() => {
      console.error = jest.fn((msg) => {
        if (msg.includes('Warning: useLayoutEffect does nothing on the server')) {
          return null;
        } else {
          originalConsoleError(msg);
        }
      });
      global.Math.random = jest.fn().mockImplementation(() => randNumber)
      global.Math.ceil = jest.fn().mockImplementation(n => n)
    });

    afterEach(() => {
      jest.resetAllMocks()
    }) 

    it('should match the snapshot', () => {
        expect(wrapper.html()).toMatchSnapshot();
      });

    it('should randomize images when click',() =>{
      wrapper.find('button[type="randomize"]').simulate('click')
      expect(wrapper.find('img').prop('alt')).toBe(`${randNumber*10}.gif`)
    })

})