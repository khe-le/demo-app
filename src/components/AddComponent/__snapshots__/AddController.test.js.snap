// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddController /> should match the snapshot 1`] = `"<div class=\\"villager-form\\"><img style=\\"height: 200px; width: auto; margin: 24px 0px;\\" src=\\"4.gif\\" alt=\\"4.gif\\"><button class=\\"btn\\" type=\\"randomize\\" style=\\"margin: 12px 0px; width: auto;\\">Show random</button></div><div class=\\"villager-form\\"><form><input class=\\"input\\" style=\\"margin: 12px 0px;\\" name=\\"firstName\\" placeholder=\\"Enter villager first name\\" values=\\"\\"><input class=\\"input\\" style=\\"margin: 12px 0px;\\" name=\\"lastName\\" placeholder=\\"Enter villager last name\\" values=\\"\\"><select class=\\"input\\" style=\\"margin: 12px 0px;\\" name=\\"job\\"><option value=\\"\\" label=\\"What the villager will do?\\"></option><option value=\\"A Nurse\\" label=\\"A Nurse\\"></option><option value=\\"A Fullstack Developer\\" label=\\"A Fullstack Developer\\"></option><option value=\\"An Artist\\" label=\\"An Artist\\"></option><option value=\\"A Witcher\\" label=\\"A Witcher\\"></option></select><button class=\\"btn\\" style=\\"margin: 12px 0px;\\" type=\\"submit\\">Add to your island</button></form></div>"`;
